README.rst

```diff
@@ -59,6 +59,26 @@ on this is available on `Cilium SBOM`_.
 .. _`SPDX`: https://spdx.dev/
 .. _`Cilium SBOM`: https://docs.cilium.io/en/latest/configuration/sbom/
 
+Development
+===========
+
+For development and testing purpose, the Cilium community publishes snapshots,
+early release candidates (RC) and CI container images build from the `master
+branch <https://github.com/cilium/cilium/commits/master>`_. These images are
+not for use in production.
+
+For testing upgrades to new development releases please consult the latest
+development build of the `Cilium Upgrade Guide <https://docs.cilium.io/en/latest/operations/upgrade/>`_.
+
+Listed below are branches for testing along with their snapshots or RC releases,
+corresponding image pull tags and their release notes where applicable:
+
++--------------------------------------------------------------------------+------------+----------------------------------------------+---------------------------------------------------------------------------------------+
+| `master <https://github.com/cilium/cilium/commits/master>`__             | daily      | ``quay.io/cilium/cilium-ci:latest``          | (No Release Notes)                                                                    |
++--------------------------------------------------------------------------+------------+----------------------------------------------+---------------------------------------------------------------------------------------+
+| `v1.14.0-snapshot.0 <https://github.com/cilium/cilium/commits/master>`__ | 2023-03-01 | ``quay.io/cilium/cilium:v1.14.0-snapshot.0`` | `Release Notes <https://github.com/cilium/cilium/releases/tag/v1.14.0-snapshot.0>`__  |
```

Comment:
Oh one more comment, this could easily break one of the currently-automated steps in `contrib/release/bump-readme.sh` once we fix it in https://github.com/cilium/cilium/pull/23954#issuecomment-1474173736 .

Reply:
(Personally I prefer to script these updates because it's way too easy to mess up the dates, or the version numbers, or the links if you're typing things manually)